typegraph/deno/src/types.ts

```diff
@@ -81,18 +100,209 @@ export function boolean(base: TypeBase = {}) {
 export class Integer extends Typedef implements Readonly<TypeInteger> {
   readonly min?: number;
   readonly max?: number;
+  readonly exclusiveMinimum?: number;
+  readonly exclusiveMaximum?: number;
+  readonly multipleOf?: number;
+  readonly enumeration?: string[];
 
   constructor(_id: number, data: TypeInteger, base: TypeBase) {
     super(_id, base);
     this.min = data.min;
     this.max = data.max;
+    this.exclusiveMinimum = data.exclusiveMinimum;
+    this.exclusiveMaximum = data.exclusiveMaximum;
+    this.multipleOf = data.multipleOf;
+    this.enumeration = data.enumeration;
   }
 }
 
 export function integer(data: TypeInteger = {}, base: TypeBase = {}) {
   return new Integer(core.integerb(data, base), data, base);
 }
 
+export class Number extends Typedef implements Readonly<TypeNumber> {
+  readonly min?: number;
+  readonly max?: number;
+  readonly exclusiveMinimum?: number;
+  readonly exclusiveMaximum?: number;
+  readonly multipleOf?: number;
+  readonly enumeration?: string[];
+
+  constructor(_id: number, data: TypeNumber, base: TypeBase) {
+    super(_id, base);
+    this.min = data.min;
+    this.max = data.max;
+    this.exclusiveMinimum = data.exclusiveMinimum;
+    this.exclusiveMaximum = data.exclusiveMaximum;
+    this.multipleOf = data.multipleOf;
+    this.enumeration = data.enumeration;
+  }
+}
+
+export function number(data: TypeNumber = {}, base: TypeBase = {}) {
+  return new Number(core.numberb(data, base), data, base);
+}
+
+export function float(data: TypeNumber = {}, base: TypeBase = {}) {
```

Comment:
I believe we may want to have a strong difference between int and float and disable the notion of number which is quite fuzzy